src/sentry/testutils/cases.py

```diff
@@ -1238,21 +1240,33 @@ def store_metric(
 
         def metric_id(key: str):
             assert isinstance(key, str)
-            res = indexer.record(use_case_id=use_case_id, org_id=org_id, string=key)
+            res = indexer.record(
+                use_case_id=REVERSE_METRIC_PATH_MAPPING[use_case_id],
```

Comment:
question: once we reach "true" use case awareness, we can't have the `REVERSE_METRIC_PATH_MAPPING` anymore - this is because each metric path key is going to map to multiple use case ids. Ultimately, these tests should also reflect that, maybe by using the actual UseCaseID param for testing. Do we feel like this should/could easily be done in this PR?

Reply:
Maybe one easy way is to just choose one of the UseCaseIDs and use it for testing -- would have to double check in the code to see if this is possible

Reply:
I agree, I think we should eliminate the usage of UseCaseKey entirely outside of metric platforms. But I think we should do this when the read path is done, because right now resolve and other functions are still using it.

Reply:
I also think this pr is already doing a lot, so we should make this change in a separate pr